src/layouts/BasicLayout.tsx

```diff
@@ -1,12 +1,17 @@
 // default imports.
 import React, { Fragment } from "react";
+import { SelectedMachine } from "../store/machineContainer";
```

Comment:
So do you wish to have a single provider?

Reply:
That will not be a good idea I think if we have a single provider and we are using only one of its property and some other property changes then all the consumers will re-render

Reply:
of course not. the stuff needed for the application on a global level goes into this store, you can store user details such as username, profile, etc in a user store. the idea is to group similar datasets and discard dissimilar ones

Reply:
Actually this machine is required in 2 components so I don't think putting it in global store is a good idea what do you think

Reply:
> of course not. the stuff needed for the application on a global level goes into this store, you can store user details such as username, profile, etc in a user store. the idea is to group similar datasets and discard dissimilar ones

Currently we don't have any such data I suppose

Reply:
i do not think it will be a problem, the global store wraps around the app globally and re-renders only the consumers. so you can use it anywhere inside the component tree

Reply:
Also, "machine store" seems too generic for the application, we also should make sure we do not create individual containers for "every" variable too.